src/pages/patterns/rich-text-editor.mdx

```diff
@@ -0,0 +1,408 @@
+
+### Responsive behavior
+
+Toolbar actions are responsive based on the width of the container. Icons can be used while keeping the chevron to show
+that there are dropdown selections.
+
+![Paragraph dropdown example where the icon replaces the label if the space is limited](figma:patterns-rich-text-editor-formatting-responsive-behavior.png)
+
+1. [Media button](/components/button#media)
+2. Media button with hidden label
+
+When a toolbar has many actions, the actions at the end will go into an overflow menu.
+
+![Overflow menu is visible if toolbar actions can't fit all the icons](figma:patterns-rich-text-editor-formatting-responsive-behavior-overflow.png)
+
+### Resizable textarea
+
+Textarea should resize vertically as the user types. The minimum and maximum height can be restricted depending on
+space constraints.
+
+![Rich-text editor with minimum height of the textarea defined](figma:patterns-rich-text-editor-formatting-resizable-textarea.png)
+
+![Rich-text editor height expands with additional content](figma:patterns-rich-text-editor-formatting-resizable-textarea-expanded.png)
+
+Once the maximum height of the textarea is reached, a scrollbar should appear next to it.
+
+![Scrollbar visible on the side once the textarea reaches its maximum height](figma:patterns-rich-text-editor-formatting-resizable-textarea-scrollbar.png)
+
+## States
+
+### Disabled states
+
+If a formatting option is not available for selected text, it should be disabled.
+
+![Disabled icon button example](figma:patterns-rich-text-editor-states-disabled.png)
+
+1. Disabled icon button
+
+## Flows
+
+### Inserting new link
+
+New link can be added by selecting a text and converting it to a link with a shortcut or by pressing the button.
+
+![Selecting text to create a link](figma:patterns-rich-text-editor-flows-insert-link-select.png)
+
+![Creating a new link from selected text](figma:patterns-rich-text-editor-flows-insert-link-save.png)
+
+### Editing link
+
+Editing link tooltip modal should give the user a choice of previewing the link and opening it in a new window,
+editing the link or breaking the link.
+
+![Previewing the link in the rich-text editor](figma:patterns-rich-text-editor-flows-edit-link-select.png)
+
+![Editing the link in the rich-text editor](figma:patterns-rich-text-editor-flows-edit-link-save.png)
+
+## Localization
+
+CKEditor offers user interface localization options and the translation system is opened to third-party plugins.
+Read more about it on the [CKEditor localization support page](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/ui/localization.html).
+
+## Accessibility
+
+### Managing focus
+
+When the toolbar is opened as a menu, the focus should be looped within that menu. This should be the case for all
+dropdowns in the toolbar.
+
+![Focus order when opening a floating toolbar](figma:patterns-rich-text-editor-accessibility-managing-focus.png)
+
+1. User selects the text they want to make bold.
```

Comment:
nitpick, remove the period to keep consistent with content guidelines

> [Do bullet points need a period?] No. Bullets should be sentence fragments. If a bullet has more than one sentence, use closing punctuation on each. If a bullet has more than two sentences, it should be a description instead.

Reply:
Good catch. Thanks!